src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -138,11 +156,32 @@ IDeviceMemoryAllocator::SAllocation CVulkanLogicalDevice::allocate(const SAlloca
         vk_allocateFlagsInfo.deviceMask = 0u; // unused: for now
     }
     VkMemoryDedicatedAllocateInfo vk_dedicatedInfo = {VK_STRUCTURE_TYPE_MEMORY_DEDICATED_ALLOCATE_INFO, nullptr};
+    VkMemoryAllocateInfo vk_allocateInfo = { VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO, &vk_allocateFlagsInfo };
+    vk_allocateInfo.allocationSize = info.size;
+    vk_allocateInfo.memoryTypeIndex = info.memoryTypeIndex;
+
+    VkImportMemoryWin32HandleInfoKHR importInfo = { 
+        .sType = VK_STRUCTURE_TYPE_IMPORT_MEMORY_WIN32_HANDLE_INFO_KHR,
+        .handleType = static_cast<VkExternalMemoryHandleTypeFlagBits>(info.externalHandleType),
+        .handle = info.externalHandle
+    };
+
+    const void** pNext = &vk_allocateFlagsInfo.pNext;
+
+    if (info.externalHandleType)
+    {
+        // Importing
+        *pNext = &importInfo;
+        pNext = &importInfo.pNext;
+    }
```

Comment:
where's the export case?

when exporting you should be patching the `CvulkanMemoryAllocation::SCreationParams::externalHandle` to contain it, so that someone can query it and then use it to import Nabla native memory to CUDA or other API themselves (outside of nabla)

Reply:
We said we wouldnt handle exports on DC

Reply:
but we need to be able to make resources-ready-for-export

i.e. for the import we handle 100% of the pipeline CUDA exp, Vulkan imp, the other way we only make the handle available, thats it, external user is on their own